### ISSUE TITLE
Unable to expand proc macros within function bodies

### ISSUE THREAD

**Issue by PizzaCrust** [intellij-rust/intellij-rust#8857]
## Environment

* **IntelliJ Rust plugin version:** 0.4.171.4630-221-nightly
* **Rust toolchain version:** 1.61.0 (fe5b13d68 2022-05-18) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2022.1.1 Community Edition (IC-221.5591.52)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Steps to reproduce
```rust
#[derive(ProcMacro)] // works
struct A;

fn b() { 
    #[derive(ProcMacro)] // broken
    struct C;
}
```
(c derivation cant be expanded, showing `internal error: expansion name not found`)

**Comment by kkolyan** [comment]
the same thing

**Comment by dima74** [comment]
Looks like it was fixed by #9280
